spec/parallel_tests/cli_spec.rb

```diff
@@ -361,6 +361,45 @@ def self.it_prints_nothing_about_rerun_commands(options)
         subject.run(['test', '-n', '3', '--only-group', '2,3', '-t', 'my_test_runner'])
       end
     end
+
+    context 'when --allow-duplicates' do
```

Comment:
There's a little bit of duplication here with the above block; that was intentional. I wanted to make sure that my changes aren't impacting existing behavior.

Once there is some comfort level with this as a solution, I'd be happy to refactor the tests to reduce some of the duplication.